pej

```diff
@@ -1,263 +1,211 @@
+
+        if result_item_date < a_week_ago:
+            continue
+        if result_item_date > end_dt:
+            continue
 
-    #main work    
-    nft_addrs = load_list(entries_file_name)
-    n = len(nft_addrs)
-    print(f"Loaded {n} NFT addrs.")
-    
-    for i, nft_addr in enumerate(nft_addrs):
-        print(f"NFT #{i+1}/{n}: Begin. nft_addr={nft_addr}")
-        pred_vals = nft_addr_to_pred_vals(nft_addr, ocean, alice) #main call
-        file_name = os.path.join(CSV_DIR, f"predvals_NFT_{nft_addr}.csv")
-        save_list(pred_vals, file_name)
-        print(f"Entry #{i+1}/{n}: done")
+        addresses.append(result_item["contractAddress"])
 
-    print("pej getpreds: Done")
+    return addresses
 
 
 @enforce_types
-def nft_addr_to_pred_vals(nft_addr:str, ocean, alice) -> List[float]:
+def nft_addr_to_pred_vals(nft_addr: str, ocean, alice) -> List[float]:
     # adapted from "What judges will do" in
     #  https://github.com/oceanprotocol/predict-eth/blob/main/challenges/main4.md
 
     # get predicted ETH values
     nft = DataNFT(ocean.config_dict, nft_addr)
     pred_vals_str_enc = nft.get_data("predictions")
-    pred_vals_str = crypto.asym_decrypt(pred_vals_str_enc, alice.private_key)
     try:
+        pred_vals_str = crypto.asym_decrypt(pred_vals_str_enc, alice.private_key)
         pred_vals = [float(s) for s in pred_vals_str[1:-1].split(',')]
     except:
-        import pdb; pdb.set_trace()
+        # import pdb; pdb.set_trace()
+        return []
 
     return pred_vals
 
-# ========================================================================
 
-@enforce_types
-def do_getnmses():
-    HELP = f"""From N predvals*.csv, output 1 nmses.csv
+def get_cex_vals(start_dt):
+    target_uts = target_12h_unixtimes(start_dt)
+    print_datetime_info("target times", target_uts)
 
-Usage: pej getnmses CSV_DIR YEAR-MONTH-DAY_HOUR:MIN
-  CSV_DIR -- input & output dir
-  YEAR:MONTH:DAY:HOUR:MIN -- start datetime, in UTC. Eg 2023-04-06_1:00 for round 4
-"""
-    if len(sys.argv) not in [2+2]:
-        print(HELP)
+    # get actual ETH values
+    cex_x = ccxt.binance().fetch_ohlcv('ETH/USDT', '1h')
+    allcex_uts = [xi[0]/1000 for xi in cex_x]
+    allcex_vals = [xi[4] for xi in cex_x]
+    # print_datetime_info("CEX data info", allcex_uts)
+    cex_vals = filter_to_target_uts(target_uts, allcex_uts, allcex_vals)
+    print(f"cex ETH price is ${cex_vals[0]} at start_dt of {start_dt}")
+    print(f"cex_vals: {cex_vals}")
+
+    return cex_vals
+
+
+def parse_arguments(arguments):
+    if len(arguments) != 4:
+        print(HELP_MAIN)
         sys.exit(0)
 
     # extract inputs
-    assert sys.argv[1] == "getnmses"
-    CSV_DIR = sys.argv[2]
-    DATETIME_STR = sys.argv[3]
-    
-    print("pej getnmses: Begin")
-    print(f"Args: CSV_DIR={CSV_DIR}, DATETIME={DATETIME_STR}")
+    assert arguments[1] == "getnmses"
+    DATETIME_STR = arguments[2]
+    ENDTIME_STR = arguments[3]
```

Comment:
Like mentioned in comments on README: CLI just needs DEADLINE. It doesn't need START_DATE. Reflect that here, and below